yaksh/views.py

```diff
@@ -2759,29 +2758,36 @@ def course_status(request, course_id):
     if not course.is_creator(user) and not course.is_teacher(user):
         raise Http404('This course does not belong to you')
     students = course.get_only_students()
+    stud_details = [(student, course.get_grade(student),
+                     course.get_completion_percent(student),
+                     course.get_current_unit(student)) for student in students]
     context = {
-        'course': course, 'students': students,
-        'state': 'course_status', 'modules': course.get_learning_modules()
+        'course': course, 'student_details': stud_details,
+        'state': 'course_status'
     }
     return my_render_to_response('yaksh/course_detail.html', context)
 
 
 def _update_unit_status(course_id, user, unit):
     """ Update course status with current unit """
-    course_status = CourseStatus.objects.filter(
+    course_status, created = CourseStatus.objects.get_or_create(
         user=user, course_id=course_id,
     )
-    if not course_status.exists():
-        course_status = CourseStatus.objects.create(
-            user=user, course_id=course_id
-        )
-    else:
-        course_status = course_status.first()
     # make next available unit as current unit
     course_status.current_unit = unit
     course_status.save()
 
 
+def _update_course_percent(course, user):
```

Comment:
Can't this be a model method?

Reply:
If I make this a model method then anyhow I have to write the Course status query every time. Instead of that, I created this view function so that this can be reused anywhere. I don't need to write course status query every time. All the other updates are being done by model methods.